### ISSUE TITLE
raising TryAgain when the retrying should be stopped causes deadlock

### ISSUE THREAD

**Issue by davidt99** [jd/tenacity#45]
I have a simple function - it checks if a request result is a certain value and if not raises `TryAgain` to try again.
I'm not sure why, but after the function should stop retrying, it enters a deadlock when `TryAgain` is raised - no new attempts are made and function doesn't return.
This simple example demonstrate it:
```python
@tenacity.retry(stop=stop_after_delay(10),wait=wait_fixed(1))
def some_function():
    raise tenacity.TryAgain()
```
I'm using python 3.5.2.

**Comment by jd** [comment]
It seems after the 10th tries, it does not return and blocks forever. Definitely a bug.

**Comment by jd** [comment]
@davidt99 fixed in 3.7.0 🤠
